mambaSharedFramework/Pantos-Generic HLS Playlist Parsing/Pantos-Generic Tag Validators/EXT_X_SESSION_DATAPlaylistValidator.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  EXT_X_SESSION_DATAPlaylistValidator.swift
+//  mamba
+//
+//  Created by Robert Galluccio on 8/31/24.
+//  Copyright © 2024 Comcast Corporation.
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License. All rights reserved.
+//
+
+import Foundation
+
+class EXT_X_SESSION_DATAPlaylistValidator: HLSPlaylistValidator {
```

Comment:
In general I prefer to use `struct` if you don't need the reference type. If you do need the reference type then `final class`

I won't repeat the comment for each validator but applies to the others as well

Reply:
Hmm... Strange... `HLSPlaylistValidator` extends `HLSValidator` that has an `AnyObject` conformance requirement:
https://github.com/Comcast/mamba/blob/develop_1.x/mambaSharedFramework/Pantos-Generic%20HLS%20Playlist%20Parsing/Pantos-Generic%20Tag%20Validators/HLSPlaylistValidator.swift#L23

So as things stand, I can't change this to a `struct`, without updating the conformance requirement and I'm not sure why that exists in the first place.

Reply:
So it's just for the playlist validator... Tag validators are fine to change to structs... I'll make the changes there.

Reply:
Updated other validators (not this one) here: c871de1e2999716474282d8d8706622ca2581dfb

Reply:
Cool- yeah I just call this out because `final class` may give some performance benefit (static vs dynamic dispatch).

Reply:
Oh, my bad! I glossed over the `final class` suggestion 🤦 

Yes, I agree that it is better to have it as `final class` rather than `class`... The static vs dynamic dispatch reasoning completely escaped me 😖 

I know you approved already, but I'll make the change anyway.

Reply:
Hmm... Actually... The only method on the class is `static` anyway... So I believe that should already be getting static dispatch optimization...

But in any case, probably more clear to move it to `final class` anyway 👍